app/parsers/caldera.py

```diff
@@ -21,9 +22,12 @@ def parse(self, report, name=None):
             return None
 
     def generate_network_map(self, report):
-        network_map = defaultdict(list)
-        report_hosts = report.hosts.keys()
-        for host in report_hosts:
-            if report.hosts[host].ports:
-                [network_map[h2].append(host) for h2 in report_hosts if h2 != host]
-        report.network_map = dict(network_map)
+        if report.network_map_nodes and report.network_map_edges:
+            return
+        network_map = nx.Graph()
+        for key,value in report.hosts.items():
+            network_map.add_node(value.ip)
```

Comment:
So what are the object types being added to the network map?

Dont we want the Host objects to be what is added to the network map?

Reply:
The way the code the was designed prior to this refactor was that the graph just showed the important info as it was pulled out of the report in pathfinder_gui.build_visualization_dataset(). Then later during adversary generation, the report is referenced for the cves, etc by using the node value (in caldera reports, that is the host IP address).

To answer your questions specifically: The object types being added are the most unique IDs for that report so the nodes can be referenced.

Since the report maintains a dictionary of {host_uuid:host_object}, we can always search the host object info with a dictionary lookup. 

Ultimately, if we want all of the node data accessible in the graph, that can be done in the pathfinder_gui.build_visualization_dataset() without having to refactor alot of code.